pypdf/_doc_common.py

```diff
@@ -1082,10 +1119,19 @@ def page_mode(self) -> Optional[PagemodeType]:
 
     def _flatten(
         self,
+        list_only: bool = False,
         pages: Union[None, DictionaryObject, PageObject] = None,
         inherit: Optional[Dict[str, Any]] = None,
         indirect_reference: Optional[IndirectObject] = None,
     ) -> None:
+        """
```

Comment:
As proposing multiline edits is a bit cumbersome, I would suggest replacing the following lines with:

```
        """
        Prepare the document pages to ease searching.

        Args:
            list_only: Will only list the pages within _flatten_pages.
            pages: 
            inherit: 
            indirect_reference: Used recursively to flatten the /Pages object.
        """
```